Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -404,11 +404,12 @@ def relim(self, visible_only=True):
             self.update_datalim(xys)
 
     def make_legend(self):
-        if self.legend_ is None:
-            legend_set_draggable(self.legend(), True)
-        else:
+        legend = self.legend_
+        if legend and legend.get_texts():
```

Comment:
I'm not sure the local variable is necessary. 
```suggestion
        if self.legend_ and self.legend_.get_texts():
```